hi-ml-cpath/src/health_cpath/utils/plots_utils.py

```diff
@@ -216,13 +219,17 @@ def __init__(
         self.should_upscale_coords = loading_params.should_upscale_coordinates()
         self.loading_params.set_roi_type_to_foreground()
         self.slides_dataset: Optional[SlidesDataset] = None
+        self.ihc_slides_dataset: Optional[SlidesDataset] = None
 
-    def get_slide_dict(self, slide_node: SlideNode) -> SlideDictType:
-        """Returns the slide dictionary for a given slide node"""
-        assert self.slides_dataset is not None, "Cannot plot attention heatmap or wsi without slides dataset"
-        slide_index = self.slides_dataset.dataset_df.index.get_loc(slide_node.slide_id)
+    def get_slide_dict(self, slide_node: SlideNode, slides_dataset: SlidesDataset) -> SlideDictType:
```

Comment:
This is now effectively a static method. You could preserve the previous usage pattern by using `slides_dataset` if passed as an argument, and otherwise using `self.slides_dataset` (just a thought, feel free to disagree!)

Reply:
The reason I made slides_dataset a param is to be able to reuse this method for the ihc_dataset as well line 258. Also Note that it's not a static method we are still using self.loading_params line 233